### DISCUSSION TITLE
How can I override default terms?

### DISCUSSION THREAD

**Discussion by orenard** [textlint/textlint#1279] [Q&A] [answered]
Hi, the default terms contain an entry that I would like to override. How can this be done? The words in question "code base", which Textlint does not like and would like to shorten to "codebase". I have tried to define my own `.textlintrc.json` containing this:

`{
  "rules": {
    "terminology": {
      "defaultTerms": false,
      "exclude": ["codebase"]
    }
  }
}
` 

But it does not have any effect.

Please note that I am using Textlint through Super Linter.

Can you please let me know how this can be done?

Many thanks

**Comment by azu** [comment]
I'm not faimilar with Super Linter, but it looks like refer to `.textlintrc` as config file.
https://github.com/super-linter/super-linter/blob/35c3fa445cc217dfcc7b53eeb4e7aa95fcdd02fc/lib/linter.sh#L208

Probably, you need to put your config to `.github/linters/.textlintrc` or specify `NATURAL_LANGUAGE_FILE_NAME=.textlintrc.json`
https://github.com/super-linter/super-linter/blob/35c3fa445cc217dfcc7b53eeb4e7aa95fcdd02fc/.github/linters/.textlintrc

**Reply by orenard** [reply]
Thank you @azu . Leaving Super Linter aside for a second and focusing solely on textlint, what can I do to get rid of these errors:

   13:62   ✓ error  Incorrect usage of the term: “code base”, use “codebase” instead  terminology
   47:22   ✓ error  Incorrect usage of the term: “code base”, use “codebase” instead  terminology
   49:27   ✓ error  Incorrect usage of the term: “code base”, use “codebase” instead  terminology
  160:199  ✓ error  Incorrect usage of the term: “code base”, use “codebase” instead  terminology

The term "code base" is valid in my use case, but the https://github.com/sapegin/textlint-rule-terminology/blob/master/terms.jsonc does not accept that spelling. How can I override that entry? I could specify `"defaultTerms": false` but this would not load all the other entries in the terms.jsonc.

I have tried this, but it does not have any effect:

```
{
  "rules": {
    "terminology": {
      "defaultTerms": true,
      "terms": [
        ["code[- ]base(es)?", "code base$1"]
      ]
    }
  }
}
```

Can you please let me know how or even if this can be done?

Many thanks

**Reply by azu** [reply, marked as answer]
I think we need to isolate whether the configuration file is not being read or whether it is an implementation problem on the rule side.
I suspect it is the latter issue, so we should wait for a response from the rule implementer.

- https://github.com/sapegin/textlint-rule-terminology/discussions/69

The issue will probably be unrelated to the textlint core.

**Reply by azu** [reply]
Having said that, after a quick look, it appears that there is an `exclude` option.
This seems to be available if you want to ignore errors.

```json
{
  "rules": {
    "terminology": {
      "defaultTerms": true,
      "exclude": ["code[- ]base(es)?"]
    }
  }
}
```

For other ways to ignore errors, see the documentation.

- https://textlint.github.io/docs/ignore.html

**Reply by orenard** [reply]
Thank you @azu . After a bit of tweaking, the exclude option does work. Thank you for your help.
